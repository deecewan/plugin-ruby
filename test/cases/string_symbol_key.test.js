runCase("string_symbol_key.rb");
runCase("string_symbol_key.rb", { preferHashLabels: false }, "hash-rockets.yml");
