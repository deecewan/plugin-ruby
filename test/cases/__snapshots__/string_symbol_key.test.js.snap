// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`string_symbol_key.rb matches expected output 1`] = `
"# frozen_string_literal: true

# rubocop:disable Lint/UnneededCopDisableDirective
# rubocop:disable Style/SymbolLiteral

def foo
  {
    'test': 1,
    'special-key': 2,
    \\"double-quote\\": 3,
    \\"#{1 + 1}\\": 4,
    \\"non-hash-label\\": 5
  }
end

# rubocop:enable Style/SymbolLiteral
# rubocop:enable Lint/UnneededCopDisableDirective
"
`;

exports[`string_symbol_key.rb matches expected output 2`] = `
"# frozen_string_literal: true

# rubocop:disable Lint/UnneededCopDisableDirective
# rubocop:disable Style/SymbolLiteral

def foo
  {
    :'test' => 1,
    :'special-key' => 2,
    :\\"double-quote\\" => 3,
    :\\"#{1 + 1}\\" => 4,
    :\\"non-hash-label\\" => 5
  }
end

# rubocop:enable Style/SymbolLiteral
# rubocop:enable Lint/UnneededCopDisableDirective
"
`;
